source/dotnet/Library/AdaptiveCards.Rendering.Html/AdaptiveCardRenderer.cs

```diff
@@ -597,27 +605,61 @@ protected static HtmlTag ColumnSetRender(AdaptiveColumnSet columnSet, AdaptiveRe
             {
                 var column = columnSet.Columns[i];
 
-                var columnRenderArgs = new AdaptiveRenderArgs(elementRenderArgs);
-                if (columnSet.Columns.Count == 1)
+                var childRenderArgs = new AdaptiveRenderArgs(childrenRenderArgs);
+
+                if (hasPadding)
                 {
-                    columnRenderArgs.ColumnRelativePosition = ColumnPositionEnum.Only;
+                    if (columnSet.Columns.Count == 1)
+                    {
+                        childRenderArgs.HasParentWithPadding = (hasPadding || parentRenderArgs.HasParentWithPadding);
+                        childRenderArgs.BleedDirection = BleedDirection.Both;
+                    }
+                    else
+                    {
+                        if (i == 0)
```

Comment:
consider creating local variables that represent the logic here to make the intent more obvious